src/NATS.Client.ObjectStore/Internal/Encoder.cs

```diff
@@ -31,7 +31,12 @@ public static string Sha256(ReadOnlySpan<byte> value)
         Span<byte> destination = stackalloc byte[256 / 8];
         using (var sha256 = SHA256.Create())
         {
+#if NETSTANDARD2_0
+            var hash = sha256.ComputeHash(value.ToArray());
+            hash.AsSpan().CopyTo(destination);
+#else
```

Comment:
Where this gets tricky is, if the object is above a certain size, grab a pooled array and copy the contents over.

-OR-, it might be better to pay the small price of something like `CommunityToolkit.HighPerformance`'s [`ReadOnlyMemory<byte>` stream adapter](https://github.com/CommunityToolkit/dotnet/blob/main/src/CommunityToolkit.HighPerformance/Streams/MemoryStream.cs) to avoid the copies.

Reply:
created https://github.com/nats-io/nats.net.v2/issues/519